synapse/handlers/federation.py

```diff
@@ -1027,6 +1012,68 @@ def _handle_new_events(self, origin, event_infos, backfilled=False,
         )
 
     @defer.inlineCallbacks
+    def _persist_auth_tree(self, auth_events, state, event):
```

Comment:
It might be nice to give a short narrative in english of what this code does.
For example explaining that it does the ``auth.check`` and explaining why do we need to ``compute_event_context`` for each event.